platform/linux_platform.go

```diff
@@ -569,6 +573,49 @@ func (p linux) SetupEphemeralDiskWithPath(realPath string) error {
 		return bosherr.WrapError(err, "Mounting data partition")
 	}
 
+	if p.options.ScrubEphemeralDisk {
+		agentVersionFilePath := path.Join(p.dirProvider.DataDir(), ".bosh", "agent_version")
+		stemcellVersionFilePath := path.Join(p.dirProvider.BaseDir(), "bosh", "etc", "stemcell_version")
+		stemcellVersion, err := p.fs.ReadFileString(stemcellVersionFilePath)
+		if err != nil {
+			return bosherr.WrapError(err, "Reading stemcell version file")
+		}
+
+		if !p.fs.FileExists(agentVersionFilePath) {
+			err = p.fs.WriteFileString(agentVersionFilePath, stemcellVersion)
+			if err != nil {
+				return bosherr.WrapError(err, "Writting agent version file")
+			}
+		} else {
+			agentVersion, err := p.fs.ReadFileString(agentVersionFilePath)
+			if err != nil {
+				return bosherr.WrapError(err, "Reading agent version file")
+			}
+
+			if agentVersion != stemcellVersion {
+				sysRunPath := path.Join(p.dirProvider.DataDir(), "sys", "run")
+				_, err = p.diskManager.GetMounter().Unmount(sysRunPath)
```

Comment:
Yes, right, if we are using a brand new disk, `/var/vcap/data/sys/run`  dir is setup after SetupEphemeralDisk() and this part of code will just write agent version file and do nothing else. I'm thinking when we are upgrading stemcell,  after mounting `/var/vcap/data`,  the directories under it have been setup as we are using the old disk, so still need to umount `/var/vcap/data/sys/run` and then delete it, right? Or am I missing something here? Thank you.